new feature add witness method (#6103)
apache/skywalking: docs/en/guides/Java-Plugin-Development-Guide.md

tomcat-7.x/8.x=TomcatInstrumentation
```

> 4. If a plugin should be activated or not in different condition.You can Override the witnessClasses or witnessMethods.

Reviewer: ```suggestion
4. Set up `witnessClasses` and/or `witnessMethods` if the instrumentation should be activated in specific versions.
```

   Example:

   ```java
   //the plugin is activated only when the foo.Bar class exists.
   @Override
   protected String[] witnessClasses() {
     return new String[] {
       "foo.Bar"
     };
   }
   
   //the plugin is activated only when the foo.Bar#hello method exists.
   @Override

>    protected List<WitnessMethod> witnessClasses() {

Reviewer: ```suggestion
   protected List<WitnessMethod> witnessMethods() {
```
Copy/paste issue?

tomcat-7.x/8.x=TomcatInstrumentation
```

4. Set up `witnessClasses` and/or `witnessMethods` if the instrumentation should be activated in specific versions.

   Example:

   ```java

>    //the plugin is activated only when the foo.Bar class exists.

Reviewer: ```suggestion
   // The plugin is activated only when the foo.Bar class exists.
```

   @Override
   protected String[] witnessClasses() {
     return new String[] {
       "foo.Bar"
     };
   }
   

>    //the plugin is activated only when the foo.Bar#hello method exists.

Reviewer: ```suggestion
   // The plugin is activated only when the foo.Bar#hello method exists.
```

   @Override
   protected List<WitnessMethod> witnessMethods() {
     List<WitnessMethod> witnessMethodList = new ArrayList<>();
     WitnessMethod witnessMethod = new WitnessMethod("foo.Bar", ElementMatchers.named("hello"));
     witnessMethodList.add(witnessMethod);
     return witnessMethodList;
   }
   ```

>    For more example, See [WitnessTest.java](/apm-sniffer/apm-agent-core/src/test/java/org/apache/skywalking/apm/agent/core/plugin/witness/WitnessTest.java)

Reviewer: ```suggestion
   For more example, see [WitnessTest.java](/apm-sniffer/apm-agent-core/src/test/java/org/apache/skywalking/apm/agent/core/plugin/witness/WitnessTest.java)
```
